code/lib/docs-tools/src/hasDocsOrControls.ts

```diff
@@ -1,7 +1,7 @@
 import type { Options } from '@storybook/types';
 
 // `addons/x` is for the monorepo, `addon-x` is for normal usage
-const packageRe = /(addons\/|addon-)(docs|controls)/;
+const packageRe = /(addons\/|addon-|addon-essentials\/)(docs|controls)/;
```

Comment:
This one had me scratching my head for half a day.  😆   I realized the Vite builder doesn't use this check, but probably should.